### ISSUE TITLE
Provide a consistent order of XML form attributes on XML form download

### ISSUE THREAD

**Issue by nmandrescu** [kobotoolbox/kpi#3061]
## Description

XML download of the form provides a random ordering of XML attributes. This makes it difficult to track in source control tools like git which are the actual changes. It will be helpful to have a consistent order of XML attributes.

## Steps to Reproduce

1. Create a form
2. Export XML
3. Update the form
4. Export XML again and compare with the previous XML.
5. Notice that besides the actual change, there are differences in unaffected places because of the attributes reordering.
6. Sometime the order doesn't change, you may need to try again.

## Expected behavior 

Define a consistent XML attributes order.

## Actual behavior 

Many times attributes are reordered, without any changes in that area.

## Additional details

Occasionally the attributes order remains the same between consecutive XML downloads.

**Comment by jnm** [comment]
Indeed! I've had to compare the resulting XML files and gotten frustrated by attributes appearing in different orders, e.g.
```xml
<bind calculate=" /data/HHMTotal " nodeset="/data/IV_RP_count" readonly="true()" type="string"/>
```
versus
```xml
<bind nodeset="/data/IV_RP_count" type="string" readonly="true()" calculate=" /data/HHMTotal "/>
```
I ended up [using](https://superuser.com/a/724436) the tool `xmllint` to standardize them. Here's an ugly one-liner that probably works only in bash:
```bash
vimdiff <(xmllint --exc-c14n first-file.xml | xmllint --format -) <(xmllint --exc-c14n second-file.xml | xmllint --format -)
```

I'm pretty certain that the attribute ordering is handled by https://github.com/XLSForm/pyxform. We're running a very old version (see #2605, #3040), so maybe with some luck upgrading will pull in a change that stabilizes the order. This looks relevant, but it seems only to apply to tests: https://github.com/XLSForm/pyxform/pull/413.

If the order remains unstable after upgrading pyxform, then this issue will have to be re-filed in the pyxform repository.
